src/VisualStudio/TestUtilities/IntegrationHelper.cs

```diff
@@ -0,0 +1,209 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Security;
+using System.Text;
+using System.Threading;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.VisualStudio.Test.Utilities.Interop;
+
+namespace Roslyn.VisualStudio.Test.Utilities
+{
+    internal static class IntegrationHelper
+    {
+        public const int DefaultTimeout = 5000; // 5 Seconds
+        public const int DefaultSleepDelay = 100; // 1/10th of a Second
+
+        private static Stopwatch _stopwatch = Stopwatch.StartNew();
+
+        public static long CurrentTickCount
+        {
+            get
+            {
+                return _stopwatch.ElapsedMilliseconds;
+            }
+        }
+
+        public static void AbortThread(Thread thread, int timeout = DefaultTimeout)
+        {
+            if ((thread != null) && (!thread.Join(timeout)))
+            {
+                try
+                {
+                    thread.Abort();
```

Comment:
`Thread.Abort` is an API that we should be avoiding.  It can cause too many issues, hangs, etc ...  Why do we need this function?  

If we get to a point where we think a thread needs to be aborted we should just cut to killing the process directly.  It's cleaner and easier to track down.